core/common/src/main/java/alluxio/cli/CommandUtils.java

```diff
@@ -104,4 +111,33 @@ public static void checkNumOfArgsNoMoreThan(Command cmd, CommandLine cl, int n)
           .getMessage(cmd.getCommandName(), n, cl.getArgs().length));
     }
   }
+
+  /**
+   * Reads a list of nodes from given file name ignoring comments and empty lines.
+   * Can be used to read conf/workers or conf/masters.
+   * @param confDir directory that holds the configuration
+   * @param fileName name of a file that contains the list of the nodes
+   * @return list of the node names, null when file fails to read
+   */
+  @Nullable
+  public static List<String> readNodeList(String confDir, String fileName) {
+    List<String> lines;
+    try {
+      lines = Files.readAllLines(Paths.get(confDir, fileName), StandardCharsets.UTF_8);
+    } catch (IOException e) {
+      System.err.format("Failed to read file %s/%s. Ignored.", confDir, fileName);
+      return new ArrayList<>();
+    }
+
+    List<String> nodes = new ArrayList<>();
+    for (String line : lines) {
+      String node = line.trim();
+      if (node.startsWith("#") || node.length() == 0) {
+        continue;
+      }
+      nodes.add(node);
+    }
+
+    return nodes;
+  }
```

Comment:
It's possible that many people don't set up their masters and workers file if they use an external system to bring up their processes. Do you think we could fall back, or cross-check with a list of masters/workers from the master to make the lives of users easier in case they don't configure their `masters` or `workers` file?

Reply:
@ZacBlanco I think we can expect users to have the masters/workers file populated for this scenario (at least for the first iteration). The idea was that this command would have the same requirements as './bin/alluxio-start.sh all'. Future versions may get smarter about this.

Reply:
@ZacBlanco the master should be able to give a list of current nodes(masters + workers) but that has downsides too. For example, if some workers are lost, do we want to collect information from those too? Also, when we want to run this command it's possible that the cluster is in an error state(or just down).

Considering the complexity and the level of correctness I think reading `conf/masters` and `conf/workers` is the easiest way. I can add more cross-checks as Zac recommends in future versions.

Reply:
Also left a TODO on `CollectInfoAll.getHosts()`

Reply:
My main use case I'm thinking of is EMR. We don't do any population of the workers files AFAIK, so it wouldn't be functional in that environment. IMO I think it should work regardless